ui/preference/list_preference.go

```diff
@@ -108,6 +108,9 @@ func (lp *ListPreferenceModal) Handle() {
 		lp.currentValue = lp.optionsRadioGroup.Value
 		lp.wallet.SaveConfigValueForKey(lp.preferenceKey, lp.optionsRadioGroup.Value)
 		lp.updateButtonClicked()
+		if lp.title == values.StrCurrencyConversion {
+			lp.IsCurrencySettingUpdated = true
+		}
 		lp.DismissModal(lp)
```

Comment:
I think we can do away with the `IsCurrencySettingUpdated` field now. `lp.DismissModal` will trigger a window refresh, which will call the main page's `Layout` method. The main page's layout method can call the `updateExchangeSetting()` method to check if the setting has changed. That method will need to be re-written like this:

```
func (mp *MainPage) updateExchangeSetting() {
	currencyExchangeValue := mp.WL.Wallet.ReadStringConfigValueForKey(dcrlibwallet.CurrencyConversionConfigKey)
	usdExchangeSet := currencyExchangeValue == values.USDExchangeValue
	if mp.usdExchangeSet == usdExchangeSet {
		return // nothing has changed
	}
	mp.usdExchangeSet = usdExchangeSet
	if mp.usdExchangeSet {
		go mp.fetchExchangeRate()
	}
}
```